scipy/signal/_peak_finding_utils.pyx

```diff
@@ -149,6 +151,11 @@ def _select_by_peak_distance(np.intp_t[::1] peaks not None,
     return keep.base.view(dtype=np.bool)  # Return as boolean array
 
 
+class PeakPropertyWarning(Warning):
```

Comment:
Maybe subclass `RuntimeWarning`? https://docs.python.org/2/library/warnings.html#warning-categories

Reply:
Great idea. `RuntimeWarning` does indeed seem like the most fitting one.